File: MovieTutorial/MovieTutorial.Web/Modules/MovieDB/MovieGenres/MovieGenresForm.cs
using Serenity.ComponentModel;

namespace MovieTutorial.MovieDB.Forms;

[FormScript("MovieDB.MovieGenres")]
[BasedOnRow(typeof(MovieGenresRow), CheckNames = true)]
public class MovieGenresForm
{
    public int MovieId { get; set; }
    public int GenreId { get; set; }
}

File: MovieTutorial/MovieTutorial.Web/Modules/ServerTypes/MovieDB/MovieGenresForm.ts
import { initFormType, LookupEditor, PrefixedContext, ServiceLookupEditor } from "@serenity-is/corelib";

export interface MovieGenresForm {
    MovieId: ServiceLookupEditor;
    GenreId: LookupEditor;
}

export class MovieGenresForm extends PrefixedContext {
    static readonly formKey = 'MovieDB.MovieGenres';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieGenresForm.init) {
            MovieGenresForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = LookupEditor;

            initFormType(MovieGenresForm, [
                'MovieId', w0,
                'GenreId', w1
            ]);
        }
    }
}